### DISCUSSION TITLE
In which scenarios a server action is accessible from client?

### DISCUSSION THREAD

**Discussion by RHOOPH** [vercel/next.js#66701] [Help]
### Summary

I have created server action by marking the file as use server;
In which of these scenarios , will NextJS expose this to the client? 

1. An exported  function  which is  only  used in server component. 
2. An exported  function that is not  used anywhere .
3. An exported function  that is used only in other server actions.
4. A function that is not exported , but used within other server actions in the same file.

 [Edited since my initial assumption of NextJS creating a unique endpoint for each server action was wrong]

**Comment by icyJoseph** [comment]
I think the server action call is made against the endpoint you are currently at, right?

**Reply by RHOOPH** [reply]
I am new to this, I thought nextjs would create unique rest like endpoint for each server action. 

Now i realise it's not the case, 

My concern was that if i create server actions like i mentioned, will the client be able to call them by making requests through postman or something?

**Reply by icyJoseph** [reply]
Yeah, you need to know a couple of things about the action you want to trigger, a header id IIRC, and include the payload necessary. 

At the end of the day, anything you make public can just be reached. And you need to validate and make sure that your handlers behave correctly.
